net/respondd/src/respondd.c

```diff
@@ -416,9 +602,33 @@ int main(int argc, char **argv) {
 				fprintf(stderr, "Multicast group must be given before interface.\n");
 				exit(EXIT_FAILURE);
 			}
+			iface_set = true;
+			last_ifindex = if_nametoindex(optarg);
 			join_mcast(sock, mgroup_addr, optarg);
 			break;
 
+		case 't':
+			if (!iface_set) {
+				fprintf(stderr, "Interface must be given before max response delay.\n");
+				exit(EXIT_FAILURE);
+			}
+
+			uint64_t max_multicast_delay = 1000 * strtoul(optarg, &endptr, 10);
```

Comment:
You must cast one of the factors to `uint64_t`, otherwise they will be multiplied as `unsigned long` and you lose the additional range provided by the `uint64_t`. The nicest way to achieve this would be to write `UINT64_C(1000)` (this macro is defined in `stdint.h`).